instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/AllocatedMemoryMetrics.java

```diff
@@ -41,9 +42,10 @@ public void bindTo(MeterRegistry registry) {
     }
 
     AllocationTracker allocationTracker = new AllocationTracker();
-    Gauge.builder("jvm.experimental.memory.allocated", allocationTracker::getAllocated)
+    Gauge.builder(METRIC_NAME, allocationTracker::getCumulativeAllocationTotal)
```

Comment:
I don't think we should add `cumulative` to the name because none of the other metrics that represent cumulative values have on. For example `jvm.gc.memory.allocated`, `db.pool.connections.timeouts`. If we decide to add something then maybe `total` instead of `cumulative`? Should we change this from gauge to counter?
@gvsmirnov would gauge -> counter be ok or would you prefer if this metic to reported delta instead?

Reply:
It is currently reporting the cumulative number of bytes allocated since JVM start. I am happy to keep the values like this. I am also fine with it becoming a counter, though the way we measure looks more like a gauge.

Reply:
Similar micrometer metrics `jvm.gc.memory.allocated` and `jvm.gc.memory.promoted` are counters. https://github.com/open-telemetry/opentelemetry-java/blob/v1.14.0/api/all/src/main/java/io/opentelemetry/api/metrics/Meter.java#L25 hints that counter might be more appropriate for metrics where one wishes to track rate of change. Though this particular piece of javadoc has been rewritten in latest version.

Reply:
Yeah, conceptually these are both counters. But in case of allocated bytes, the way we measure them does not quite allow us to do `counter.add(delta)` and there is no `counter.set()` in the API as far as I can see.

Reply:
We could try
```java
    FunctionCounter.builder("jvm.experimental.memory.allocated", allocationTracker, (unused) -> allocationTracker.getAllocated())
        .description("Approximate sum of heap allocations")
        .baseUnit(BaseUnits.BYTES)
        .register(registry);
```
I wouldn't be too concerned about how this looks in the agent but rather how backend handles these. Like counters and gauges having different rollup. Unfortunately I don't know enough of the topic to say with certainty how these should be handled.

Reply:
I think the backend does NOT offer the "rate" **rollup** on gauges, but so far we've been using a "rate of change" **transformation** and it worked. But if we can use a counter that would be great